Packs/GenericWebhook/Integrations/GenericWebhook/GenericWebhook.py

```diff
@@ -0,0 +1,74 @@
+from io import StringIO
+from traceback import format_exc
+from typing import Dict
+
+import uvicorn
+from fastapi import FastAPI, Request, Response
+from pydantic import BaseModel
+
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+
+class Incident(BaseModel):
+    name: Optional[str] = None
+    type: Optional[str] = None
+    occurred: Optional[str] = None
+    raw_json: Optional[Dict] = None
+
+
+app = FastAPI(logger=DebugLogger(), docs_url=None, redoc_url=None, openapi_url=None)
+
+
+@app.middleware('http')
+async def verify_auth_header(request: Request, call_next):
+    auth_header = demisto.params().get('auth_header')
+    if auth_header and request.headers.get('Authorization') != auth_header:
+        demisto.debug(f'Authorization failed - request headers {request.headers}')
+        return Response(status_code=401, content='Authorization failed.')
+    return await call_next(request)
+
+
+@app.post('/')
+async def handle_post(incident: Incident, request: Request) -> List:
+    raw_json = incident.raw_json or await request.json()
+    return demisto.createIncidents([{
+        'name': incident.name or 'Generic webhook triggered incident',
+        'type': incident.type or demisto.params().get('incidentType'),
+        'occurred': incident.occurred,
+        'rawJSON': json.dumps(raw_json)
+    }])
+
+
+def main() -> None:
+    demisto.debug(f'Command being called is {demisto.command()}')
+    try:
+        default_log_stream = None
+        try:
+            port = int(demisto.params().get('longRunningPort'))
+        except ValueError as e:
+            raise ValueError(f'Invalid listen port - {e}')
+        if demisto.command() == 'test-module':
+            return_results('ok')
+        elif demisto.command() == 'long-running-execution':
+            while True:
+                try:
+                    default_log_stream = StringIO()
+                    log_config = uvicorn.config.LOGGING_CONFIG
+                    log_config['handlers']['default']['stream'] = default_log_stream
```

Comment:
I actually think we should enable the logging handler. It helps debugging all kinds of issues. You can try using: `DemistoHandler`. I guess you will need to add the following line:
```
log_config['handlers']['default']['class'] = 'DemistoHandler`
```
also I would enable the access_log.

Reply:
Or try using `IntegrationLogger()` instead of StringIO.

Reply:
i initialize the app logger to be `DebugLogger()` so i think its redundant so set up the logging handler here as well, but i might miss something
lets discuss